pvdeg/utilities.py

```diff
@@ -416,3 +416,72 @@ def ts_gid_df(file, gid):
                 res.lat = meta.latitude[gid]
                 res.lon = meta.longitude[gid]
     return res
+
+
+def tilt_azimuth_scan(
+    weather_df=None,
+    meta=None,
+    weather_kwarg=None,
+    tilt_count=18,
+    azimuth_count=72,
+    sky_model="isotropic",
+    temp_model="sapm",
+    conf_0="insulated_back_glass_polymer",
+    conf_inf="open_rack_glass_polymer",
+    T98=70, # [°C]
+    x_0=6.5, # [cm]
+    wind_speed_factor=1.7):
+
+    """
+    Calculate a minimum standoff distance for roof mounded PV systems as a function of tilt and azimuth.
+
+    Parameters
+    ----------
+    weather_df : pd.DataFrame
+        Weather data for a single location.
+    meta : pd.DataFrame
+        Meta data for a single location.
+    tilt_count : integer
+        Step in degrees of change in tilt angle of PV system between calculations.
+    azimuth_count : integer
+        Step in degrees of change in Azimuth angle of PV system relative to north.
+    sky_model : str, optional
+        Options: 'isotropic', 'klucher', 'haydavies', 'reindl', 'king', 'perez'.
+    temp_model : str, optional
+        Options: 'sapm'.  'pvsyst' and 'faiman' will be added later.
+        Performs the calculations for the cell temperature.
+    conf_0 : str, optional
+        Default: 'insulated_back_glass_polymer'
+    conf_inf : str, optional
+        Default: 'open_rack_glass_polymer'
+    x_0 : float, optional
+        Thermal decay constant [cm], [Kempe, PVSC Proceedings 2023]
+    wind_speed_factor : float, optional
+        Wind speed correction factor to account for different wind speed measurement heights
+        between weather database (e.g. NSRDB) and the tempeature model (e.g. SAPM)
+        The NSRD provides calculations at 2m (i.e module height) but SAPM uses a 10m height.
+        It is recommended that a power-law relationship between height and wind speed of 0.33 be used.
+        This results in a wind_speed_factor of 1.7. It is acknowledged that this can vary significantly. 
+    Returns
+        standoff_series : 2-D array with each row consiting of tilt, azimuth, then standoff
+    """
+    
+    standoff_series=np.array([(azimuth_count+1)*(tilt_count+1)][3])
```

Comment:
invalid index

Reply:
This is just incomplete and should not be imported. I apologize for not following protocol as I'm getting a larger understanding of what we are doing. I'll be more professional going forward as I'm figuring this out.

Reply:
no problem at all. you are doing a great job for your first pull request. I just want to make sure everything is working as expected.

Reply:
there are a couple of ways, what you can do to not include code that's not ready for a PR. 

1) instead of committing it to the branch you are working on, you can commit it to a new branch for the development of the feature. the new branch can be made directly form the one you were working on. So, all the other changes are still in there. 

2) you could use git stash before you commit. this way, the code snipped remains on your computer, you can finish up the other work, and then when you'll keep working on it you can bring it back. here is some more explanation: https://medium.com/vscodetips/understanding-git-stash-and-using-it-easily-in-vscode-7da049160097

Reply:
Yes, lets just postpone this addition. I'd have to check but at most there might be a small documentation change somewhere. While programming, I got to this point and then thought it was a good spot to get everything accepted because I think the important things that have broad impact had been done. This addition, is just an additional capability not affecting prior code.

Reply:
Just learning a bit of the purpose, procedure, and philosophy of the different stages of code modification.  I'll make better, more logical decisions about the stages for committing changes going forward.